cms/djangoapps/api/docs/decisions/001-course-apps-api.rst

```diff
@@ -0,0 +1,152 @@
+
+A GET request to this API will return an array of objects with the above
+structure. A PATCH request to the same endpoint with just the `id` and the
+`enabled` attribute can be used to enable/disable the app if the user has the
+requisite permissions.
+
+Each app will also have a user-friendly title and description, however this will
+be defined in the frontend since that is where internationalisation will happen.
+The frontend can look up the title and description based on the app's `id`.
+
+This data will be provided by a plugin configuration that is part of the app.
+For old/existing apps, we can create plugin configurations that calls back to
+existing code.
+
+To maintain consistency around how a course app is enabled/disabled, we can use
+Course Waffle Flags as the mechanism for enabling/disabling a course app for a
+particular course or globally. We can create a new Waffle namespace called
+"course_apps", and the course app ID can be the name for the `CourseWaffleFlag`
+for each plugin by convention.
+
+To enumerate the list of available apps, we will list the installed plugins,
+check which plugins are available for the current course, and call each plugin
+to get its metadata. Each plugin will support checking if the plugin is
+available for a particular course. If the plugin marks itself as unavailable, it
+won't be listed in the API response.
+
+Only legacy apps will support the `legacy_link`, it will not be a publicised
+part of the API so new/external plugin apps shouldn't use it. This link should
+only be provided for course apps that don't have a UI in the course authoring
+MFE. If a partial UI exists, the MFE settings view can always link back to the
+old studio view from there.
+
+The following is a sample class with the pattern for such plugins:
+
+.. code-block :: python
+
+    WIKI_ENABLED = CourseWaffleFlag(COURSE_APPS_WAFFLE_NAMESPACE, 'wiki', __name__)
+
+    class CourseApp:
+        app_id: str = 'wiki'
+
+        IS_AVAILABLE = WIKI_ENABLED
+
+        # This method will not be in the sample/base class, but will be added to
+        # existing course apps.
+        if LEGACY_APP:
+            @classmethod
+            def legacy_link(cls, course_key):
+                return f'some/link/to/{course_key}'
+
+        @classmethod
+        def is_enabled(cls, course_key):
+            # Some logic to check if the app is enabled for this course
+            # This will not vary from user-to-user in studio.
+            return True
+
+        @classmethod
+        def get_permissions(cls, course_key, user):
+            # This should return a dictionary with at least an `enable` key.
+            return {
+                'enable': can_user_enable(course_key, user),
+                'configure': can_user_configure(course_key, user),
+                'edit_lti_config': is_user_admin(course_key, user),
+            }
+
+
+For existing course apps, we can create such a class and have these class
```

Comment:
```suggestion
For existing course apps, we can create such classes and have these class
```

Reply:
I think this one is OK since it's "a class" for each course app. I've clearified it.